python/tests/kernel/test_kernel_features.py

```diff
@@ -559,16 +566,113 @@ def kernel(params: np.ndarray):
     assert len(counts) == 2 and '01' in counts and '10' in counts
 
 
-def test_capture_vars_disallowed():
+def test_capture_vars():
 
     n = 5
+    f = 0.0
+    m = 5
+    hello = str()
 
     @cudaq.kernel
-    def test():
+    def kernel():
         q = cudaq.qvector(n)
+        x(q)
+        cudaq.dbg.ast.print_f64(f)
+        for qb in q:
+            rx(f, qb)
+
+    counts = cudaq.sample(kernel)
+    counts.dump()
+    assert '1' * n in counts
+
+    f = np.pi
+    counts = cudaq.sample(kernel)
+    counts.dump()
+    assert '0' * n in counts
+
+    n = 7
+    f = 0.0
+    counts = cudaq.sample(kernel)
+    counts.dump()
+    assert '1' * n in counts
+
+    counts = cudaq.sample(kernel)
+    counts.dump()
+    assert '1' * n in counts
+
+    n = 3
+    counts = cudaq.sample(kernel)
+    counts.dump()
+    assert '1' * n in counts
```

Comment:
Should there be an assert about the length change (`n`) here?

Reply:
Probably a good idea - I was thinking it was implicit since `'1' * n -> '111'` check (if this PR did not work, it should fail since '111' is not in counts, but '11111' is)